src/plugins/remoteClock/RemoteClock.js

```diff
@@ -152,15 +152,18 @@ export default class RemoteClock extends DefaultClock {
    */
   #waitForReady() {
     const waitForInitialTick = (resolve) => {
-      if (this.lastTick > 0) {
-        const offsets = this.openmct.time.getClockOffsets();
-        resolve({
-          start: this.lastTick + offsets.start,
-          end: this.lastTick + offsets.end
-        });
-      } else {
-        setTimeout(() => waitForInitialTick(resolve), 100);
-      }
+      const tickListener = () => {
+        if (this.lastTick > 0) {
+          const offsets = this.openmct.time.getClockOffsets();
+          this.openmct.time.off('tick', tickListener); // Unregister the tick listener
+          resolve({
+            start: this.lastTick + offsets.start,
+            end: this.lastTick + offsets.end
+          });
+        }
+      };
+
+      this.openmct.time.on('tick', tickListener);
```

Comment:
You can use `this.openmct.time.once` and remove the boilerplate code that removes the listener above.

Reply:
i _think_ that is correct, but didn't do it because if for some reason there is a non positive tick. Is that possible?

Reply:
In our test environments a negative time should never occur. It's not invalid in theory though, which is _WHY WE SHOULD NEVER USER 0 AS A DEFAULT VALUE FOR TIME_.

Reply:
It's fine, let's leave it as-is.